File: java/Visitor/src/Customer.java
package Exercise;

import java.util.ArrayList;

public class Customer implements _______________ {// <<1

	String name;

	private ArrayList<Order> orders = new ArrayList<Order>();

	public void accept(Visitor visitor) {
		visitor._______________(this);// <<2	
		for (Order eachOrder: orders){
			eachOrder._______________(visitor);// <<3
		}
	}

	public void addOrder(Order order) {
		orders.add(order);
	}

	public Customer(String name) {
		this.name = name;
	}
}

File: ruby/Visitor/customer.rb
require_relative 'visitable'
require_relative 'visitor'
require_relative 'order'

class Customer < _______________ # <<1
  attr_accessor :name
  def initialize(name)
    @name = name
    @orders = Array.new
  end

  def accept(visitor)
    visitor._______________(self) #<<2
    for order in @orders
      order._______________(visitor) # <<3
    end
  end

  def add_order(order)
    @orders.push(order)
  end
end